torch/distributed/distributed_c10d.py

```diff
@@ -74,8 +77,14 @@ def __new__(cls, name):
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
             raise ValueError("Invalid backend: '{}'".format(name))
+        elif value != Backend.GLOO and value != Backend.NCCL and value != Backend.MPI:
+            value, _ = getattr(Backend, name.upper())
         return value
 
+    @classmethod
+    def register_new_backend(cls, module_name, instantiating_interface_name):
+        setattr(Backend, module_name.upper(), (module_name, instantiating_interface_name))
```

Comment:
What is about to split module name and backend name?

E.g. python module with new transport will have some complex name like "torch_xxx_yyy_zzz" and user will import module by this name, but will pass some short name to init_process_group(backend='aaa', ...).

Otherwise currently user should pass again full module name as backend name.

Reply:
I update the code to support your idea. now as long as user register a short backend name in register_backend(), then it will be used to match with the one passed in init_process_group().